components/event-publisher-proxy/pkg/handler/nats_test.go

```diff
@@ -0,0 +1,205 @@
+package handler
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/env"
+	"github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/legacy-events"
+	"github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/options"
+	"github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/receiver"
+	"github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/sender"
+	"github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/subscribed"
+	testingutils "github.com/kyma-project/kyma/components/event-publisher-proxy/testing"
+	eventingv1alpha1 "github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha1"
+	eventingtesting "github.com/kyma-project/kyma/components/eventing-controller/testing"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic/dynamicinformer"
+	dynamicfake "k8s.io/client-go/dynamic/fake"
+)
+
+func TestNatsHandler(t *testing.T) {
+	t.Parallel()
+
+	logger := logrus.New()
+	logger.Info("TestNatsHandler started")
+
+	// a cancelable context to be used
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	port, err := testingutils.GeneratePort()
+	if err != nil {
+		t.Fatalf("failed to generate port: %v", err)
+	}
+	cfgNats := newEnvConfig(port)
+
+	// configure message receiver
+	messageReceiver := receiver.NewHttpMessageReceiver(cfgNats.Port)
+	assert.NotNil(t, messageReceiver)
+
+	// Start Nats server
+	natsServer := eventingtesting.RunNatsServerOnPort(cfgNats.Port + 1)
+	assert.NotNil(t, natsServer)
+	defer natsServer.Shutdown()
+
+	// create a Nats sender
+	natsUrl := natsServer.ClientURL()
+	assert.NotEmpty(t, natsUrl)
+	sender := sender.NewNatsMessageSender(ctx, natsUrl, logger)
```

Comment:
`sender` variable collides with import package(`"github.com/kyma-project/kyma/components/event-publisher-proxy/pkg/sender"`). Let's change this to something else.

Reply:
Done